heat/core/logical.py

```diff
@@ -547,3 +549,88 @@ def signbit(x: DNDarray, out: Optional[DNDarray] = None) -> DNDarray:
     DNDarray([False,  True, False], dtype=ht.bool, device=cpu:0, split=None)
     """
     return _operations.__local_op(torch.signbit, x, out, no_cast=True)
+
+
+def in1d(
+    x: Union[DNDarray, int, float],
+    y: Union[DNDarray, int, float],
+    assume_unique: Optional[bool] = False,
+    invert: Optional[bool] = False,
+) -> DNDarray:
+    """
+    Test whether each element of a 1-D array is also present in a second array.
+    Returns a boolean array the same length as `x` that is True
+    where an element of `x` is in `y` and False otherwise.
+    We recommend using :func:`isin` instead of `in1d` for new code.
+
+    Parameters
+    ----------
+    x : DNDarray or scalar
+        Input array.
+    y : DNDarray or scalar
```

Comment:
We should implement `isin`, and then alias `in1d` to a special case of `isin`.

We can use the generic  `_operations.binary_op` function with `torch.isin` as local function, there's no need to implement the algorithm from scratch.